itests/hive-minikdc/src/test/java/org/apache/hive/minikdc/TestAuthWithDigestMD5.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.token.Token;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+public class TestAuthWithDigestMD5 {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TestAuthWithDigestMD5.class.getName());
+    protected static HiveMetaStoreClient client;
+    public static final String HS2TOKEN = "HiveServer2ImpersonationToken";
+
+    protected static String correctUser = "correct_user";
+    protected static String correctPassword = "correct_passwd";
+
+    protected static Configuration conf = null;
+
+    private static MiniHiveKdc miniKDC = null;
+    protected static Configuration clientConf;
+    protected static String hiveMetastorePrincipal;
+    protected static String hiveMetastoreKeytab;
+    private static boolean isServerStarted = false;
+    protected static int port;
+
+    @Before
+    public void setUp() throws Exception {
+        miniKDC = new MiniHiveKdc();
+        hiveMetastorePrincipal =
+                miniKDC.getFullyQualifiedServicePrincipal(miniKDC.getHiveMetastoreServicePrincipal());
+        hiveMetastoreKeytab = miniKDC.getKeyTabFile(
+                miniKDC.getServicePrincipalForUser(miniKDC.getHiveMetastoreServicePrincipal()));
+
+        if (null == conf) {
+            conf = MetastoreConf.newMetastoreConf();
+        }
+
+        MetastoreConf.setBoolVar(conf, ConfVars.EXECUTE_SET_UGI, false);
+        clientConf = new Configuration(conf);
+
+        MetastoreConf.setVar(conf, ConfVars.THRIFT_METASTORE_AUTHENTICATION, "CONFIG");
+        MetastoreConf.setVar(conf, ConfVars.THRIFT_AUTH_CONFIG_USERNAME, correctUser);
+        MetastoreConf.setVar(conf, ConfVars.THRIFT_AUTH_CONFIG_PASSWORD, correctPassword);
+        MetastoreConf.setBoolVar(conf, ConfVars.USE_THRIFT_SASL, true);
+        MetastoreConf.setVar(conf, ConfVars.KERBEROS_PRINCIPAL, hiveMetastorePrincipal);
+        MetastoreConf.setVar(conf, ConfVars.KERBEROS_KEYTAB_FILE, hiveMetastoreKeytab);
+
+        // set some values to use for getting conf. vars
+        MetastoreConf.setBoolVar(conf, ConfVars.METRICS_ENABLED, true);
+        conf.set("datanucleus.autoCreateTables", "false");
+        conf.set("hive.in.test", "true");
+        MetastoreConf.setVar(conf, ConfVars.METASTORE_METADATA_TRANSFORMER_CLASS, " ");
+
+        MetaStoreTestUtils.setConfForStandloneMode(conf);
+        MetastoreConf.setLongVar(conf, ConfVars.BATCH_RETRIEVE_MAX, 2);
+        MetastoreConf.setVar(conf, ConfVars.STORAGE_SCHEMA_READER_IMPL, "no.such.class");
+        MetastoreConf.setBoolVar(conf, ConfVars.INTEGER_JDO_PUSHDOWN, true);
+        MetastoreConf.setTimeVar(conf, ConfVars.DELEGATION_TOKEN_RENEW_INTERVAL,10000, TimeUnit.MILLISECONDS);
+        MetastoreConf.setTimeVar(conf, ConfVars.DELEGATION_TOKEN_GC_INTERVAL,3000, TimeUnit.MILLISECONDS);
```

Comment:
Instead of default set separate warehouse dir, not to clash with other warehouse dirs

Reply:
This is also not required, in this case while starting the metastore, the warehouse directory will be port specific. so even if multiple instances of HMS are started on different ports in this case the warehouse dir will be different.